### DISCUSSION TITLE
Events missing

### DISCUSSION THREAD

**Discussion by nulkathos** [blakeblackshear/frigate#9405] [Ask A Question]
I'm checking if everythings is working fine and the system is able to detect everywhere what will be on the cam.
 I've found that on the yellow area...events will be created...but no one on the red area..
the red area is far 39 meters from the camera.
![298295838-89ffd69c-672c-4a72-97c9-494afdace3e4-0](https://github.com/blakeblackshear/frigate/assets/155397425/559ee227-25f3-4cfd-9861-d7d8297bc2d8)

I've tried to use a debug view as suggested:
![298370375-972e41f5-721b-402e-99eb-abb57012dee8-1](https://github.com/blakeblackshear/frigate/assets/155397425/aee5df4e-409a-4532-b759-1ae48c65f676)

events was creted/recorded only when the subject reached the yellow area:
![298370473-9bac71b0-48a2-43e6-8e08-415a7ca6f946-2](https://github.com/blakeblackshear/frigate/assets/155397425/b7bc8854-2d11-4fe0-a7bd-8b90a7f010e0)

following my config (I'm using the main stream for detect and record):
![image](https://github.com/blakeblackshear/frigate/assets/155397425/81ac71a9-a24b-40a0-8559-992dbd474aeb)

that is my camera config:
![image](https://github.com/blakeblackshear/frigate/assets/155397425/4e207675-ac6d-4624-b6b8-c1555706def6)

How can be fixed?

**Comment by NickM-27** [comment]
a video of the debug view would be more helpful but in any case it is not clear what boxes were enabled, `bounding boxes` would be the most interesting to see

**Reply by nulkathos** [reply]
attached  a debug view video:

https://github.com/blakeblackshear/frigate/assets/155397425/6b843ccd-b640-4545-b886-098a4a8c469a

**Reply by NickM-27** [reply]
The model is most likely just not detecting in that scenario or the scores are lower than the default min_score. Other users had to submit examples of people further away to get it through improve as well so it's also possible that the updated base model will work better in this way.

**Reply by nulkathos** [reply]
ok, I'll give more images

**Comment by jaaneo** [comment]
The detection does not need to be in 4k or main stream, according to the frigate documentation, the recommended stream for detection should be the sub_stream.

Main Stream (Recording & RTSP)

Encode Mode: H.264
Resolution: 2688*1520
Frame Rate(FPS): 15
I Frame Interval: 30
Sub Stream (Detection)

Enable: Sub Stream 2
Encode Mode: H.264
Resolution: 1280*720
Frame Rate: 5
I Frame Interval: 5

https://docs.frigate.video/frigate/camera_setup

Maybe if you configure it this way it will recognize the event you mention.

**Reply by nulkathos** [reply]
it was configured this way before I've seen that no events was creato so I moved to a higher resolution for detect.
Documentation also is talking about: "Larger resolutions do improve performance if the objects are very small in the frame."
@blakeblackshear @NickM-27 : there are improvements or will be consumed compute resources for a very little improvements?

**Reply by NickM-27** [reply]
can you open http://frigate_ip:5000/api/ovestnord/grid.jpg and paste that here

**Reply by nulkathos** [reply]
![image](https://github.com/blakeblackshear/frigate/assets/155397425/8e0e5072-1cdd-4047-8d6e-6d52a5079ca5)

**Reply by NickM-27** [reply]
If you're able, I'd suggest stopping frigate and then running sqlite3 `DELETE FROM regions;`

**Reply by nulkathos** [reply]
not really. which is the right command inside the frigate container to stop the service?
docker exec -it {containername} sh -c " command"

**Reply by NickM-27** [reply]
you would basically want to shut down frigate then edit the file from the host by running `sqlite3 /path/to/frigate.db`

**Reply by nulkathos** [reply]
ok so stopping the container and modifying the frigate-compose.yml
![image](https://github.com/blakeblackshear/frigate/assets/155397425/5d1e93ed-dd6c-4215-9fad-ebf4c07663f0)
editing the volume /media/frigate?

**Reply by NickM-27** [reply]
yeah you would run `sqlite3 /media/frigate/frigate.db` and then once that opens enter `DELETE FROM regions;` and then hit enter.

**Reply by jaaneo** [reply]
/api/name_camera/grid.jpg fits all?
I get a Not Found

**Reply by NickM-27** [reply]
You must be on 0.13

**Reply by nulkathos** [reply]
ok just to be sure that I've understood....I need to install sqlite3(is not installed on the vm) and exec the command on the machine for the file frigate.db that you pasted?

**Reply by NickM-27** [reply]
yes

**Reply by NickM-27** [reply]
might want to make a backup to be safe

**Reply by nulkathos** [reply]
backup done.
regions deleted.
now the result is:
![image](https://github.com/blakeblackshear/frigate/assets/155397425/955d1fe1-ac6c-4ee1-8a4e-5462418af8f9)

**Reply by NickM-27** [reply]
good, now when you start frigate with the higher detect resolution we may expect to see some improvement

**Reply by nulkathos** [reply]
I'll give an update about the detection at high distances... but I have a question...
changing resolution also had or will have impact on the model? 
I'm asking that because, training again (I'm now near 1k images, the first was at around 800 ) seems that is not improved...seems like I was starting today to use it.....antenna/horse/tree stomp... are detected like "person"
![image](https://github.com/blakeblackshear/frigate/assets/155397425/dc584910-0785-42d4-b5d0-e31c12bdb9a0)
an event is in progress (about a tree stomp) from 5 hours...
and after many false positive that I've sent also
![image](https://github.com/blakeblackshear/frigate/assets/155397425/cc2ed906-4a0f-4e09-bce5-3054ff481a2f)

**Reply by NickM-27** [reply]
The resolution won't affect that since the regions for training are downsized to the model size. 

Some users have found some false positives to be more difficult to train out than others. I imagine false positives like this are not common in the current base model but an updated base model may perform better along with you continuing to submit false positive examples. 

I'd also suggest motions making the timestamp if you haven't (this does change with resolution)

**Reply by blakeblackshear** [reply]
When you run detection at high resolutions to detect smaller objects, then you are always going to increase the potential for false positives. There are so many more places for those false positives to be since its analyzing every small detail in the entire frame to see if its an expected object type. Many of these events are <90% confidence, so you can at least reduce the quantity by updating your filters and focusing Frigate+ submissions on higher scores. I know you probably lowered them to detect small objects, so there is a tradeoff here.

**Comment by nulkathos** [comment]
I've tested many times and I can say that is working...it's detecting small object... 💯 
![image](https://github.com/blakeblackshear/frigate/assets/155397425/20393410-08af-4849-bf6f-08239f223671)

I've requested a training a couple of days ago, adding around 700 events to the base, where 90% false positives(my animals detected as a person)  . 
was a try to reduce the false positives because the next week I will add 6 more cam ( and don't be overwhelmed by hundred of events )

may be the system/model needs something like 10K of Images to be trustable ( I don't know) but after the last training I cannot say if something was improved:
![image](https://github.com/blakeblackshear/frigate/assets/155397425/79c2cb88-966b-4cb5-aefd-551e6e274fea)

**Reply by NickM-27** [reply]
it looks like you have not increased your threshold like the docs recommend. 71% is too low of a threshold for the frigate+ model unless you are specifically trying to get examples to submit (but your description sounds like you want to avoid these https://docs.frigate.video/plus/first_model#step-4-adjust-your-object-filters-for-higher-scores

**Reply by nulkathos** [reply]
In my config there are no filters
with frigate+ was suggested not use it or I haven't understood correctly?

**Reply by NickM-27** [reply]
there should be min_score and threshold filters when using frigate, like the docs I linked above suggest

**Reply by nulkathos** [reply]
added
![image](https://github.com/blakeblackshear/frigate/assets/155397425/0dcdc4b6-373e-4f11-93c8-78c0a35b4e3b)

**Reply by nulkathos** [reply]
ok, with filters some events was not recorded ( but horses with >90% I think  it's difficult)

but an event was missed today..

![image](https://github.com/blakeblackshear/frigate/assets/155397425/e8ea823e-1766-453b-bab1-3dc6e25bea06)
the record terminated with :
![image](https://github.com/blakeblackshear/frigate/assets/155397425/e6b0dab0-84af-4248-ac63-50e6fb6f1c09)
where the person is still present...
the recording process ....stopped and no more events...so...what is the process?? has to create a new event...or maybe continuing to register on the same event ?

**Reply by NickM-27** [reply]
if an object stops being detected and is detected again then a new event will be created.

**Reply by nulkathos** [reply]
ok...it's only strange because...

in my config:
![image](https://github.com/blakeblackshear/frigate/assets/155397425/2ed584f6-5446-4163-bcce-8222646fc8c2)
so...with 7s precapture....it wa detected on
![image](https://github.com/blakeblackshear/frigate/assets/155397425/bbc0b74c-11df-4763-8783-7af39848d7f5)
weare saying that
![image](https://github.com/blakeblackshear/frigate/assets/155397425/ba2cf6f6-0ad4-480e-95f5-9e66f3044d38)
was better detected than 
![image](https://github.com/blakeblackshear/frigate/assets/155397425/95f9d563-01af-4b0e-adfc-2ce0cf51e4b5)

**Reply by NickM-27** [reply]
Pre capture doesn't work like that. Pre capture extends the time of the event but it doesn't perfectly line up with the recordings because the recordings are in 10 second segments.
